File: backend/semgrep_rules/tests/python/kisa_sw_001_good.py
import sqlite3

from django.http import HttpRequest


def find_user(
    request: HttpRequest
):

    user_id = (
        request.GET.get(
            "id"
        )
    )


    connection = (
        sqlite3.connect(
            "test.db"
        )
    )


    cursor = (
        connection.cursor()
    )


    query = (
        "SELECT * "
        "FROM users "
        "WHERE id = ?"
    )


    cursor.execute(
        query,
        (
            user_id,
        )
    )


    result = (
        cursor.fetchall()
    )


    connection.close()


    return result

File: backend/semgrep_rules/tests/java/kisa_sw_001_good.java
import java.sql.Connection;
import java.sql.PreparedStatement;
import javax.servlet.http.HttpServletRequest;

class SqlInjectionGood {

    public void search(
        HttpServletRequest request,
        Connection connection
    ) throws Exception {

        String id =
            request.getParameter("id");

        String query =
            "SELECT * FROM users WHERE id = ?";

        PreparedStatement statement =
            connection.prepareStatement(
                query
            );

        statement.setString(
            1,
            id
        );

        statement.executeQuery();
    }
}